src/components/ScenarioParameters/components/ScenarioActions/ScenarioActions.spec.js

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { ButtonTesting } from '../../../../../tests/MuiComponentsTesting';
+import { customRender } from '../../../../../tests/utils/renderInThemeAndStoreProviders';
+import { LaunchButton, SaveButton, DiscardChangesButton, StopRunButton } from './components/';
+import { getByDataCy } from '../../../../../tests/utils';
+import { SCENARIO_RUN_STATE } from '../../../../services/config/ApiConstants';
+import { useFormState } from 'react-hook-form';
+import { useCurrentScenarioState } from '../../../../state/hooks/ScenarioHooks';
+
+jest.mock('react-hook-form', () => ({
+  __esModule: true,
+  ...jest.requireActual('react-hook-form'),
+  useFormState: jest.fn(),
+}));
+
+jest.mock('@cosmotech/ui', () => ({
+  __esModule: true,
+  ...jest.requireActual('@cosmotech/ui'),
+  PermissionsGate: ({ children }) => {
+    return <div data-testid="PermissionsGate">{children}</div>;
+  },
+}));
+
+const mockUseLaunchScenario = jest.fn();
+const mockUseSaveAndLaunchScenario = jest.fn();
+const mockUseSaveScenario = jest.fn();
+const mockUseStopScenarioRun = jest.fn();
+
+jest.mock('../../../../state/hooks/ScenarioHooks', () => ({
+  __esModule: true,
+  ...jest.requireActual('../../../../state/hooks/ScenarioHooks'),
+  useLaunchScenario: () => mockUseLaunchScenario,
+  useSaveAndLaunchScenario: () => mockUseSaveAndLaunchScenario,
+  useSaveScenario: () => mockUseSaveScenario,
+  useCurrentScenarioState: jest.fn(),
+}));
+
+jest.mock('../../../../state/hooks/ScenarioRunHooks', () => ({
+  __esModule: true,
+  ...jest.requireActual('../../../../state/hooks/ScenarioRunHooks'),
+  useStopScenarioRun: () => mockUseStopScenarioRun,
+}));
+
+jest.mock('../../../../hooks/ScenarioParametersHooks', () => ({
+  __esModule: true,
+  ...jest.requireActual('../../../../hooks/ScenarioParametersHooks'),
+  useUpdateParameters: () => ({
+    forceUpdate: false,
+    processFilesToUpload: jest.fn(),
+    getParametersToUpdate: jest.fn(),
+  }),
+}));
+
+const launchScenarioButton = new ButtonTesting({ dataCy: 'launch-scenario-button' });
+const saveScenarioButton = new ButtonTesting({ dataCy: 'save-button' });
+const discardChangesButton = new ButtonTesting({ dataCy: 'discard-button' });
+const stopRunButton = new ButtonTesting({ dataCy: 'stop-scenario-run-button' });
+
+describe('Test scenario buttons when scenario is not running', () => {
+  beforeAll(() => {
+    useCurrentScenarioState.mockReturnValue(() => null);
+  });
+
+  describe('Test scenario buttons if form is not dirty', () => {
+    beforeAll(() => {
+      useFormState.mockReturnValue({ isDirty: false });
+      customRender(<LaunchButton />);
```

Comment:
Why do you render each buttons instead of ScenarioActions ?
in this case, the tests should be placed in each button component, like SaveButton.spec.js.